siliconcompiler/core.py

```diff
@@ -3369,12 +3374,17 @@ def _runtask(self, step, index, active, error):
 
         all_inputs = []
         if not self.get('remote'):
+            #copy over status from
+
             for in_step, in_index in self.get('flowgraph', flow, step, index, 'input'):
                 index_error = error[in_step + in_index]
                 self.set('flowstatus', in_step, in_index, 'error', index_error)
                 if not index_error:
                     cfgfile = f"../../../{in_job}/{in_step}/{in_index}/outputs/{design}.pkg.json"
-                    self.read_manifest(cfgfile, clobber=False)
+                    with open(cfgfile, 'r') as f:
+                        localcfg = json.load(f)
+                    # quick copy of task history
+                    self.cfg['metric'][in_step][in_index] = copy.deepcopy(localcfg['metric'][in_step][in_index])
```

Comment:
I think this might be missing some stuff we need to carry forward:

- `'eda', ... 'report'`
- `'record', ...`
- `'flowstatus', ...` (less sure about this one, but is that the reason maybe it seemed like you needed to hack up some stuff related to flowstatus elsewhere?)

Reply:
Agree on the 'record', that is definitely missing. Let's fix the flowstatus when you update the run schedulibg and modular flowgraphs. They all go together.